sys/net/gnrc/pktshark/gnrc_pktshark.c

```diff
@@ -0,0 +1,721 @@
+/*
+ * Copyright (C) 2025 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_gnrc_pktshark
+ * @{
+ *
+ * @file
+ * @brief       Pretty-printer for packages sent/received via netapi
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <inttypes.h>
+#include <stdio.h>
+
+#include <errno.h>
+#include "auto_init_utils.h"
+#include "auto_init_priorities.h"
+#include "byteorder.h"
+#include "fmt.h"
+#include "thread.h"
+#include "msg.h"
+#include "net/gnrc/pktdump.h"
+#include "net/gnrc.h"
+#include "net/icmpv6.h"
+#include "net/ipv6/addr.h"
+#include "net/ipv6/hdr.h"
+#include "net/sixlowpan/nd.h"
+#include "net/tcp.h"
+#include "net/udp.h"
+#include "net/sixlowpan.h"
+#include "net/nanocoap.h"
+#include "net/ipv4/hdr.h"
+#include "od.h"
+#include "shell.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/**
+ * @brief   PID of the pktshark thread
+ */
+static kernel_pid_t gnrc_pktshark_pid = KERNEL_PID_UNDEF;
+
+/**
+ * @brief   Stack for the pktshark thread
+ */
+static char _stack[GNRC_PKTDUMP_STACKSIZE];
+static msg_t _msg_queue[GNRC_PKTDUMP_MSG_QUEUE_SIZE];
+
+/* Iterator for NDP options in a packet */
+#define FOREACH_OPT(ndp_pkt, opt, icmpv6_len) \
+    for (opt = (ndp_opt_t *)(ndp_pkt + 1); \
+         icmpv6_len > 0; \
+         icmpv6_len -= (opt->len << 3), \
+         opt = (ndp_opt_t *)(((uint8_t *)opt) + (opt->len << 3)))
+
+static void _dump_rtr_sol(const ndp_rtr_sol_t *rtr_sol, size_t payload_len)
+{
+    print_str("Router Sol");
+
+    ndp_opt_t *opt;
+    FOREACH_OPT(rtr_sol, opt, payload_len) {
+        switch (opt->type) {
+        case NDP_OPT_SL2A:
+            print_str(" sl2a");
```

Comment:
Why not output the content of this as well?

Reply:
I didn't want to clutter the output too much